src/main/fc/rc.c

```diff
@@ -496,134 +496,182 @@ NOINLINE void initAveraging(uint16_t feedforwardAveraging)
 
-    // calculate acceleration and attenuate
-    setpointAcceleration = (setpointSpeed - prevSetpointSpeed[axis]) * rxRate * 0.01f;
+    // calculate setpointAcceleration from smoothed setpoint speed
+    setpointAcceleration = (setpointSpeed - prevSetpointSpeed[axis]);
     prevSetpointSpeed[axis] = setpointSpeed;
 
-    // smooth the acceleration element (effectively a second order filter) and apply jitter reduction
+    // smooth the acceleration element (effectively a second order filter since incoming setpoint was already smoothed)
     setpointAcceleration = prevAcceleration[axis] + pid->feedforwardSmoothFactor * (setpointAcceleration - prevAcceleration[axis]);
-    prevAcceleration[axis] = setpointAcceleration * zeroTheAcceleration;
-    setpointAcceleration = setpointAcceleration * pid->feedforwardBoostFactor * jitterAttenuator * zeroTheAcceleration;
+    prevAcceleration[axis] = setpointAcceleration;
+
+    // apply gain factor to acceleration
+    setpointAcceleration *= rxRate * pid->feedforwardBoostFactor;
+
+    feedforward = setpointSpeed;
+
+    if (axis < FD_YAW) {
+        // for pitch and roll, add acceleration element (boost)
+        feedforward += setpointAcceleration;
+        // apply jitter reduction multiplier
+        feedforward *= jitterAttenuator;
+        // pull feedforward back towards zero as sticks approach max
+        if (pid->feedforwardMaxRateLimit) {
+            if (feedforward * setpoint > 0.0f) { // in same direction
+                const float limit = (maxRcRate[axis] - fabsf(setpoint)) * pid->feedforwardMaxRateLimit;
+                feedforward = (limit > 0.0f) ? constrainf(feedforward, -limit, limit) : 0.0f;
+            }
+        }
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 1, lrintf(setpointSpeed * 0.1f)); // base feedforward without acceleration
+    } else {
+        // for yaw, only apply jitter reduction only to the feedforward delta element
+        // can't be applied to the 'sustained' element since it will divide it to a small fraction of what is needed
+        feedforward *= jitterAttenuator;
+
+        // instead of adding setpoint acceleration, which is too aggressive,
+        // add a slow-fading high-pass filtered setpoint element
+        // this provides a 'sustained boost' with low noise
+        // it mimics the normal sustained yaw motor drive requirements, reducing P and I and hence reducing bounceback
+        // this doesn't add significant noise to feedforward
+        // too little yaw FF causes iTerm windup and slow bounce back when stopping a hard yaw
+        // too much causes fast bounce back when stopping a hard yaw
+
+        static float prevSetpointYaw;
+        // calculate lowpass gain factor from user specified time constant, can't exceed 1.0
+        const float kFY = fminf(rxInterval * pid->feedforwardYawHoldTime, 1.0f);
```

Comment:
To get the gain from the time constant, you need to use `pt1FilterGainFromDelay();`
https://github.com/betaflight/betaflight/blob/master/src/main/common/filter.c#L54-L63

Reply:
Thanks, the `pt1FilterGainFromDelay` function is now done, that's great.
I'm having difficulty doing the `pt1Filter_t` approach.
I can add `static pt1Filter_t feedforwardYawHoldLpf;` at the top where the other feedforward statics go.
And where we use the filter, `const float setpointLpfYaw = pt1FilterApply(& feedforwardYawHoldLpf, setpoint);`, that's OK too.
But... we don't have an 'init' for feedforward in rc.c, and hence I'm not sure where to put the line `pt1FilterInit(& feedforwardYawHoldLpf, 0.0f);` (assuming I can init it to 0) - where or how should I most efficiently init the filter?

Reply:
The RC codebase is a messy nightmare. There is a lot of initialization done in the runtime part. Ideally, you would make a `rcInit()` function, where all the initializations from this file go into. Then run it once in the function `init()` in the file `init.c`. That will make sure everything RC related is initialized on BF startup.

Reply:
I've made a simple init and implemented the 'proper' method, I think.
Pls see most recent commit, not sure if it's correctly done.
It adds about 40 bytes to F411 code compared to the previous commit.
What should we do.

Reply:
That is still the "old" way without a proper initialization routine, which initializes stuff in the runtime part. A clean solution would introduce a function that runs all RC related initializations once on BF startup in init.c but this is fine for now. `rc.c` is already super messy, so don't worry too much.

Reply:
OK the above changes didn't work, when I tested them.  I'll leave the old fashioned manual way for now; perhaps later you can do a 'real good fix' :-) ?